### PR TITLE
Add API type conversion

### PR DESCRIPTION
**Pull request by jhernand**
Currently the rules specified in the configuration use explicitly version `v1alpha2` of the API. This means that if in the future we introduce and support a new version of the API we will have to stop supporting `v1alpha2`, or else explicitly convert those types to the new version of the API. To improve that this patch uses the Kubernetes API version conversion mechanisms to deserialize the rules. That way the user will be able to specify in the configuration file the version of each rule. For example:

```yaml
rules:

- apiVersion: autoheal.openshift.io/v1alpha2
  kind: HealingRule
  metadata:
    name: my-old-rule
  ...

- apiVersion: autoheal.openshift.io/v1
  kind: HealingRule
  metadata:
    name: my-new-rule
  ...
```

The default values for `apiVersio` will be `autoheal.openshift.io/v1alpha2` so users don't need to explicitly indicate it.

### CODE REVIEW

**Comment by jhernand** [comment]
@zgalor @yaacov @moolitayer @nimrodshn @elad661 please review.

File: pkg/apis/autoheal/v1alpha2/zz_generated.conversion.go [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+// Code generated by conversion-gen. DO NOT EDIT.
+
+// This file was autogenerated by conversion-gen. Do not edit it manually!
+
+package v1alpha2
+
+import (
+	unsafe "unsafe"
+
+	autoheal "github.com/openshift/autoheal/pkg/apis/autoheal"
```
Comment (yaacov) [inline pkg/apis/autoheal/v1alpha2/zz_generated.conversion.go]: why do we need to use explicit package names here ?
can we just do ?
```
import (
    "unsafe"
    "github.com/openshift/autoheal/pkg/apis/autoheal"        
    ...
```

what am i missing here ?

File: pkg/apis/autoheal/register.go [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+	AddToScheme   = SchemeBuilder.AddToScheme
+)
+
+// Adds the list of known types to the given scheme.
+//
+func addKnownTypes(scheme *runtime.Scheme) error {
+	scheme.AddKnownTypes(
+		SchemeGroupVersion,
+		&HealingRule{},
+		&HealingRuleList{},
```
[the remark this answers is not in the document]
Comment (jhernand) [reply pkg/apis/autoheal/register.go]: No, this should only contain the top level types. It is like in the Kubernetes API: Pod is a top level type but PodSpec isn't.

**Pull request merged**